pytorch_lightning/core/memory.py

```diff
@@ -389,3 +379,46 @@ def get_human_readable_count(number: int) -> str:
     number = number * (10 ** shift)
     index = num_groups - 1
     return f"{int(number):,d} {labels[index]}"
+
+
+def _format_summary_table(total_params: int, total_trainable_params: int, *cols) -> str:
+    """
+    Takes in a number of arrays, each specifying a column in
+    the summary table, and combines them all into one big
+    string defining the summary table that are nicely formatted.
+    """
+    n_rows = len(cols[0][1])
+    n_cols = 1 + len(cols)
+
+    # Get formatting width of each column
+    col_widths = []
+    for c in cols:
+        col_width = max(len(str(a)) for a in c[1]) if n_rows else 0
+        col_width = max(col_width, len(c[0]))  # minimum length is header length
+        col_widths.append(col_width)
+
+    # Formatting
+    s = "{:<{}}"
+    total_width = sum(col_widths) + 3 * n_cols
+    header = [s.format(c[0], l) for c, l in zip(cols, col_widths)]
+
+    # Summary = header + divider + Rest of table
+    summary = " | ".join(header) + "\n" + "-" * total_width
+    for i in range(n_rows):
+        line = []
+        for c, l in zip(cols, col_widths):
+            line.append(s.format(str(c[1][i]), l))
+        summary += "\n" + " | ".join(line)
+    summary += "\n" + "-" * total_width
+
+    # Biggest text below is "Non-trainable params"
+    params_summary_margin = len("Non-trainable params") + 2
+    summary += "\n" + " " * params_summary_margin + "Totals"
+    summary += "\n" + s.format("Trainable params", params_summary_margin)
+    summary += get_human_readable_count(total_trainable_params)
+    summary += "\n" + s.format("Non-trainable params", params_summary_margin)
+    summary += get_human_readable_count(total_params - total_trainable_params)
+    summary += "\n" + s.format("Total params", params_summary_margin)
+    summary += get_human_readable_count(total_params)
```

Comment:
Moved the function below since it now needs the definition of `get_human_readable_count` function.

Reply:
but it doesn't matter, right?